src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,650 @@
+    private static final String DEFAULT_ROW_INFLUENCE = "0";
+
+    /** Specify nearby text pattern to trigger filter to begin suppression. */
+    private Pattern nearbyTextPattern = Pattern.compile(DEFAULT_NEARBY_TEXT_PATTERN);
+
+    /**
+     * Specify check name pattern to suppress. Property can also be a RegExp group index
+     * at {@code nearbyTextPattern} in format of {@code $x} and be picked from target file.
+     */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String checkPattern = DEFAULT_CHECK_PATTERN;
+
+    /** Specify check violation message pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String messagePattern;
+
+    /** Specify check ID pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String idPattern;
+
+    /**
+     * Specify negative/zero/positive value that defines the number of lines
+     * preceding/at/following the suppressing nearby text. Property can also be a RegExp group
+     * index at {@code nearbyTextPattern} in format of {@code $x} and be picked from target file.
+     */
+    private String rowInfluence = DEFAULT_ROW_INFLUENCE;
+
+    /**
+     * Setter to specify nearby text pattern to trigger filter to begin suppression.
+     *
+     * @param pattern a {@code Pattern} value.
+     */
+    public final void setNearbyTextPattern(Pattern pattern) {
+        nearbyTextPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check name pattern to suppress. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setRowInfluence(String format) {
+        rowInfluence = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+        if (event.getViolation() != null) {
+            final FileText fileText = getFileText(event.getFileName());
```

Comment:
@romani We are reloading the file directly in this new filter. Is this ok?

Reply:
Better to avoid reloading.

Reply:
@stoyanK7 We are going to make this a separate issue. Audit Events and filters don't collect the file source, so your filter won't have any way to work without doing what your doing now.

Reply:
This conversation can be resolved.